cpp/open3d/core/nns/FaissIndex.cpp

```diff
@@ -109,16 +109,15 @@ std::pair<Tensor, Tensor> FaissIndex::SearchKnn(const Tensor &query_points,
     int64_t num_query_points = query_points.GetShape()[0];
     knn = std::min(knn, (int)GetDatasetSize());
 
-    auto *data_ptr = static_cast<const float *>(query_points.GetDataPtr());
+    auto *data_ptr = query_points.GetDataPtr<float>();
```

Comment:
Could you double-check if we actually need/not-need the `const`? 
- Here the `const` is missing but in other locations, I see that we still keep the `const`.
- We also have two getters already: `const T* GetDataPtr()` and `T* GetDataPtr()`, can the `const` be somehow determined automatically by the compiler?

Reply:
This is True. The const can be automatically determined by the compiler. I've removed the const when using GetDataPtr